tests/filecheck/lit.cfg

```diff
@@ -1,25 +1,34 @@
 import lit.formats
 import os
+from pathlib import Path
 
 config.test_source_root = os.path.dirname(__file__)
-xdsl_src = os.path.dirname(os.path.dirname(config.test_source_root))
+xdsl_src = Path(config.test_source_root).parent.parent
+
+tools_dir = xdsl_src / "xdsl" / "tools"
+xdsl_opt = tools_dir / "xdsl_opt.py"
+xdsl_run = tools_dir / "xdsl_run.py"
+irdl_to_pyrdl = tools_dir / "irdl_to_pyrdl.py"
 
 config.name = "xDSL"
-config.test_format = lit.formats.ShTest(preamble_commands=[f"cd {xdsl_src}"])
 config.suffixes = ['.test', '.mlir', '.py']
 
-xdsl_opt = "xdsl/tools/xdsl_opt.py"
-xdsl_run = "xdsl/tools/xdsl_run.py"
-irdl_to_pyrdl = "xdsl/tools/irdl_to_pyrdl.py"
-
 config.substitutions.append(('XDSL_ROUNDTRIP', "xdsl-opt %s --print-op-generic --split-input-file | xdsl-opt --split-input-file | filecheck %s"))
 config.substitutions.append(("XDSL_GENERIC_ROUNDTRIP", "xdsl-opt %s --print-op-generic --split-input-file | filecheck %s --check-prefix=CHECK-GENERIC"))
+
 if "COVERAGE" in lit_config.params:
-    config.substitutions.append(('xdsl-opt', f"coverage run {xdsl_opt}"))
-    config.substitutions.append(('xdsl-run', f"coverage run {xdsl_run}"))
-    config.substitutions.append(('irdl-to-pyrdl', f"coverage run {irdl_to_pyrdl}"))
-    config.substitutions.append(('python', f"coverage run"))
+    coverage_cmd = f"coverage run"
+    config.test_format = lit.formats.ShTest(preamble_commands=[f"cd {xdsl_src}"])
+else:
+    rc_file = xdsl_src / "pyproject.toml"
+    data_file = xdsl_src / ".coverage.filecheck"
+    coverage_cmd = f"coverage run --source={xdsl_src} --rcfile={rc_file} --data-file={data_file}"
+    config.test_format = lit.formats.ShTest()
 
+config.substitutions.append(('xdsl-opt', f"{coverage_cmd} {xdsl_opt}"))
+config.substitutions.append(('xdsl-run', f"{coverage_cmd} {xdsl_run}"))
+config.substitutions.append(('irdl-to-pyrdl', f"{coverage_cmd} {irdl_to_pyrdl}"))
+config.substitutions.append(('python', f"{coverage_cmd}"))
```

Comment:
Basically I couldn't hack making the change work with coverage enabled, so we now have a bit more logic that depends on whether or not we collect code coverage metrics, such as where the tests are run from, and what the paths to the commands are.

Reply:
Maybe this is worth a comment?